src/main/java/org/traccar/WebDataHandler.java

```diff
@@ -211,7 +219,11 @@ public String formatRequest(Position position) throws UnsupportedEncodingExcepti
 
         private void send() {
             if (json) {
-                requestBuilder.async().post(Entity.json(payload), this);
+                try {
+                    requestBuilder.async().post(Entity.entity(objectMapper.writeValueAsString(payload), mediaType), this);
+                } catch (JsonProcessingException e) {
+                    throw new RuntimeException("Failed to serialize payload to json: " + payload);
```

Comment:
I don't think we need to append payload here. It's too long if it's the content or doesn't make sense if it's just a reference.